### PR TITLE
Unbind all WS events on close 

### PR DESCRIPTION
**Pull request by akirilyuk**

### CODE REVIEW

**Comment by akirilyuk** [comment]
@tasoili Hey Tass, addd the tests here.

File: test/transport_test.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+        }
+    }
+    mockery.enable({
+        warnOnReplace: false,
+        warnOnUnregistered: false,
+        useCleanCache: true
+    });
+    mockery.registerMock('ws', Websocket);
+    const Transport = require('./../lib/Transport');
```
Comment (tasoili) [inline test/transport_test.js:24]: Unless there's a reason this is particularly important, could you move this to be with the other `require`s above?

File: test/transport_test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+'use strict';
+const expect = require('chai').expect;
+const mockery = require('mockery');
+const sinon = require('sinon');
+const Events = require('events');
```
Comment (tasoili) [inline test/transport_test.js:5]: Is there a reason this is uppercase? I also noticed this in Transport.js.

File: test/transport_test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        warnOnUnregistered: false,
+        useCleanCache: true
+    });
+    mockery.registerMock('ws', Websocket);
+    const Transport = require('./../lib/Transport');
+    let transport;
+    beforeEach(()=>{
+        wsConstructorSpy.reset();
+        transport = new Transport({});
+    });
```
Comment (tasoili) [inline test/transport_test.js:29]: Please add blank lines between all these blocks to match our other files.

**Review by tasoili — APPROVED** [review]
Looks good! Just a couple of small things for you to change, if you could.

File: test/transport_test.js (hunk 3 above)
Comment (akirilyuk) [reply test/transport_test.js:29]: ok

File: test/transport_test.js (hunk 2 above)
Comment (akirilyuk) [reply test/transport_test.js:5]: its a class, so the notation should be camelcase.

File: test/transport_test.js (hunk 1 above)
Comment (akirilyuk) [reply test/transport_test.js:24]: if you require it before registering the mock, the mocking won't work...

**Pull request merged**